miner/worker.go

```diff
@@ -1113,8 +1124,14 @@ func (w *worker) commitWork(interrupt *int32, noempty bool, timestamp int64) {
 	if !noempty && atomic.LoadUint32(&w.noempty) == 0 {
 		w.commit(work.copy(), nil, false, start)
 	}
+
 	// Fill pending transactions from the txpool
-	w.fillTransactions(interrupt, work)
+	err = w.fillTransactions(interrupt, work)
+	if err != nil && !errors.Is(err, errBlockInterruptedByRecommit) {
```

Comment:
This is perhaps a bit more clear? 
```suggestion
	if errors.Is(err, errBlockInterruptedByNewHead) {
```

Reply:
I think the original expresses the intent more clearly - only recommit interrupted blocks should be considered.
If this is important to you I'll change it but otherwise I'd leave it as is.

Reply:
I disagree. Your intent:

> Head-interrupted blocks should not be commit-ed

Therefore, it seems to me that checking `errors.Is(err, errBlockInterruptedByNewHead)` directly translates to the intented effect, as opposed to `err != nil && !errors.Is(err, errBlockInterruptedByRecommit)`

Reply:
I agree with martin.